File: src/main/java/com/HOIIVUtils/clauzewitz/data/technology/Technology.java
package com.HOIIVUtils.clauzewitz.data.technology;

import com.HOIIVUtils.clauzewitz.code.ClausewitzDate;
import com.HOIIVUtils.clauzewitz.localization.Localizable;
import org.jetbrains.annotations.NotNull;

import java.util.Collection;
import java.util.List;
import java.util.Map;

public class Technology implements Localizable {
	public String id;
	public ClausewitzDate technologyDate;
	public double cost;    // research_cost
	public int year;    // start_year
	public List<Equipment> enabledEquipments;
	public List<EquipmentModule> enabledEquipmentModules;
	public List<TechCategory> categories;
	public List<TechPath> paths;
	// ai_will_do

	@Override
	public @NotNull Map<Property, String> getLocalizableProperties() {
		return Map.of(
			Property.NAME, id
		);
	}

	// todo this can be more inclusive
	@Override
	public @NotNull Collection<? extends Localizable> getLocalizableGroup() {
		return List.of(this);
	}
}

File: src/main/scala/com/hoi4utils/hoi4mod/common/technologies/Technology.scala
package com.hoi4utils.hoi4mod.common.technologies

import com.hoi4utils.hoi4mod.localization.{Localizable, Property}
import com.hoi4utils.parser.ClausewitzDate
import com.hoi4utils.parser.ClausewitzDate
import org.jetbrains.annotations.NotNull

import scala.collection.mutable


class Technology extends Localizable {
	var id: String = ""
	var technologyDate: Option[ClausewitzDate] = None
	var cost = .0 // research_cost

	var year = 0 // start_year

	var enabledEquipments: List[Equipment] = Nil
	var enabledEquipmentModules: List[EquipmentModule] = Nil
	var categories: List[TechCategory] = Nil
	var paths: List[TechPath] = Nil

	// ai_will_do
	@NotNull override def getLocalizableProperties: mutable.Map[Property, String] = mutable.Map(Property.NAME -> id)

	// todo this can be more inclusive
	@NotNull override def getLocalizableGroup: Iterable[? <: Localizable] = List(this)
}